app/services/census_authorization_handler.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+# Checks the authorization against the census for Barcelona.
+class CensusAuthorizationHandler < Decidim::AuthorizationHandler
+  include ActionView::Helpers::SanitizeHelper
+
+  attribute :document_number, String
+  attribute :document_type, String
+  attribute :postal_code, String
+  attribute :date_of_birth, Date
+
+  validates :document_number, :document_type, :postal_code, :date_of_birth,
+            presence: true
+  validates :document_type, inclusion: { in: %w{dni nie passport} }
+
+  # The only method that needs to be implemented for an authorization handler.
+  # Here you can add your business logic to check if the authorization should
+  # be created or not, you should return a Boolean value.
+  def authorized?
+    valid? && response.xpath("//codiRetorn").text == "01"
+  end
+
+  # If you need to store any of the defined attributes in the authorization you
+  # can do it here.
+  #
+  # You must return a Hash that will be serialized to the authorization when
+  # it's created, and available though authorization.metadata
+  def metadata
+    super.merge(postal_code: postal_code)
```

Comment:
No @josepjaume, unless you specify it nothing gets saved. We shouldn't store the document number right?